kroxylicious-filters/kroxylicious-encryption/src/test/java/io/kroxylicious/filter/encryption/dek/AtomicCryptorUsageCounterTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.dek;
+
+import org.assertj.core.api.AbstractBooleanAssert;
+import org.junit.jupiter.api.Test;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class AtomicCryptorUsageCounterTest {
```

Comment:
With this change do we end up with duplicate coverage in `DekTest`? Or is that operating at a different level of abstraction and thus its worth validating that it collaborates  correctly with the `AtomicCryptorUsageCounter`

Reply:
I think we need to ensure we're calling the right mutators in the right places, so lets not tear up those tests in `DekTest`.

Reply:
yeahp given this counter a primitive I don't think there's a useful way to mock it out. Dek also adds in some useful bits like ensuring we can't close a Cryptor twice causing a double decrement.